mesh/DMesh3.cs

```diff
@@ -240,17 +246,23 @@ public CompactInfo CompactCopy(DMesh3 copy, bool bNormals = true, bool bColors =
 
             // [TODO] would be much faster to explicitly copy triangle & edge data structures!!
 
+            int[] mapT = new int[copy.MaxTriangleID];
+            for (int i = 0; i < mapT.Length; i++)
+                mapT[i] = IndexMap.InvalidIndex;
             foreach ( int tid in copy.triangles_refcount ) {
                 Index3i t = copy.GetTriangle(tid);
                 t.a = mapV[t.a]; t.b = mapV[t.b]; t.c = mapV[t.c];
                 int g = (copy.HasTriangleGroups) ? copy.GetTriangleGroup(tid) : InvalidID;
-                AppendTriangle(t, g);
+                mapT[tid] = AppendTriangle(t, g);
+                if (mapT[tid] < 0)
```

Comment:
from performance point of view - it makes sense to put TriangleId into an Int variable, to avoid reading it from array.

Reply:
Added a variable